oracle/oracle.go

```diff
@@ -1211,6 +1219,72 @@ func (or *Oracle) handleManualUnsubscriptions(
 	}
 }
 
+func (or *Oracle) handleManualBans(
+	banEvents []*contract.ContractBanValidator) {
+
+	// FIRST: healthy checks, ensure the bans events are okay.
+
+	// Ensure the bans events are from the same block
+	if len(banEvents) > 0 {
+		blockReference := banEvents[0].Raw.BlockNumber
+		for _, ban := range banEvents {
+			if ban.Raw.BlockNumber != blockReference {
+				log.Fatal("Handling manual bans from different blocks is not possible: ",
+					ban.Raw.BlockNumber, " vs ", blockReference)
+			}
+		}
+	}
+
+	totalPending := big.NewInt(0)
+	// SECOND: iterate over the ban events.
+	//  - Advance state machine of all banned validators (move them to Banned state).
+	//  - Sum all the pending rewards of the banned validators and share them among the rest.
+
+	for _, ban := range banEvents {
+		log.WithFields(log.Fields{
+			"BlockNumber":    ban.Raw.BlockNumber,
+			"TxHash":         ban.Raw.TxHash,
+			"ValidatorIndex": ban.ValidatorID,
+		}).Info("[Ban] Ban event received")
+		or.advanceStateMachine(ban.ValidatorID, ManualBan)
```

Comment:
Can you add a test here for what happens if the ban.ValidatorId doesn't exit? Unsure if the current behaviour is what we want or will crash.

same for unban.